include/envoy/http/filter.h

```diff
@@ -189,6 +189,12 @@ class StreamDecoderFilterCallbacks : public virtual StreamFilterCallbacks {
    */
   virtual const Buffer::Instance* decodingBuffer() PURE;
 
+  /**
+   * Allows modifying the decoding buffer. May only be called before any data has been continued
+   * past the calling filter.
+   */
+  virtual void modifyDecodingBuffer(std::function<void(Buffer::Instance&)> callback) PURE;
```

Comment:
Sorry, chiming in late.

If we're going to need a custom filter in order to buffer the whole body and then call modifyDecodingBuffer, would it be possible to refactor the buffering filter to have a stream complete callback and folks can subclass and implement the onBufferingFilterStreamComplete callback?   Or better yet we could implement #5834, subclass the buffering filter and override the base class onEncode/DecodeComplete.

My concern is both adding extra complexity to the HCM for something I think we can push into an exiting filter, and I think by design it's subject to the data-with-end-stream problem called out below.

Reply:
This is a good point. @snowp if this makes sense to you I would be in favor of closing this and doing what @alyssawilk says?